tests/NexusMods.App.GarbageCollection.DataModel.Tests/FindUsedFiles/MarkGameFilesTest.cs

```diff
@@ -0,0 +1,129 @@
+using NexusMods.StandardGameLocators.TestHelpers.StubbedGames;
+
+namespace NexusMods.App.GarbageCollection.DataModel.Tests.FindUsedFiles;
+
+/// <summary>
+/// This ensures that 'game files' are marked as roots when performing the GC action.
+/// That is, the backup of game data which we made.
+/// </summary>
+/// <param name="serviceProvider"></param>
+public class MarkGameFilesTest(IServiceProvider serviceProvider) : GCStubbedGame(serviceProvider)
+{
+    [Fact]
+    public async Task GameFilesAreRooted()
+    {
+        // Setup: Manage a game and make a 'vanilla' loadout.
+        // This will run the synchronizer, and thus in turn ingest the tested file from GCStubbedGame
+        await CreateLoadout();
+
+        // As a sanity test, confirm that we have backed up the test file.
+        // This proves our initial assertion that synchronizer runs as expected
+        (await FileStore.HaveFile(ExpectedHash)).Should().Be(true);
+        
+        // Act: Run a GC.
+        var gc = CreateGC();
+        var newArchivePath = RunGarbageCollector(gc, out var toDelete);
+
+        // Assert: No game files should be deleted from FileStore, they are roots.
+        (await FileStore.HaveFile(ExpectedHash)).Should().Be(true);
+        
+        // Unmanage the game (with GC)
+        await Synchronizer.UnManage(GameInstallation, runGc: true);
+        
+        // The file should be gone, because we deleted/retracted all the roots.
+        (await FileStore.HaveFile(ExpectedHash)).Should().Be(false);
+    }
+    
+    private AbsolutePath RunGarbageCollector(ArchiveGarbageCollector<NxParsedHeaderState, FileEntryWrapper> collector, out List<Hash> toDelete)
+    {
+        AbsolutePath newArchivePath = default;
+        List<Hash> toDel = null!;
+        collector.CollectGarbage(new Progress<double>(), (progress, toArchive, toRemove, archive) =>
+            {
+                toDel = toRemove;
+                NxRepacker.RepackArchive(progress, toArchive, toRemove, archive, true, out newArchivePath);
+            }
+        );
+
+        toDelete = toDel;
+        return newArchivePath;
+    }
+    
+    private ArchiveGarbageCollector<NxParsedHeaderState, FileEntryWrapper> CreateGC()
+    {
+        // Note: This ignores stubbed game files, we're not testing for those.
+        var gc = new ArchiveGarbageCollector<NxParsedHeaderState, FileEntryWrapper>();
+        DataStoreReferenceMarker.MarkUsedFiles(Connection, gc); // <= picks up our marker on GcRootFileName
+        return gc;
+    }
+    
+    public class Startup
+    {
+        // https://github.com/pengweiqhca/Xunit.DependencyInjection?tab=readme-ov-file#3-closest-startup
+        // A trick for parallelizing tests with Xunit.DependencyInjection
+        public void ConfigureServices(IServiceCollection services) => DIHelpers.ConfigureServices(services);
+    }
+}
+
+/// <summary>
+/// A stubbed game for GC testing.
+/// </summary>
+public class GCStubbedGame : AGameTest<StubbedGame>
```

Comment:
Is it possible to use `AIsolatedGameTest` here instead? Not sure on the differences in detail, but we have been using the isolated version for new tests since it was redone with the purpose of avoiding parallel execution issues.

Reply:
I can change it, but it wouldn't make a difference, I still need a `Startup` object for XUnit DI.

Reply:
That said, I'd need to write some additional boilerplate somewhere it seems, to handle getting the stubbed game into the GameRegistry, it's not showing up there.

Reply:
@Al12rs I had to add some missing functionality, notably the 'AIsolatedGameTest` did not support the `GenerateGameFiles` API. But I did it just to make you happy.

It seems all of the code there was duplicated and slightly tweaked in `AIsolatedGameTest`. I wonder why this couldn't just have been a bool flag into the constructor of `AGameTest`, we shouldn't really be duplicating code like this :sweat_smile: . Not sure how that wound up all the way in the codebase.

There isn't a difference here at runtime though, I had to add a tiny bit of extra boilerplate I didn't have before actually to make this work now.

Reply:
I think the plan to to delete AGameTest when we don't have any uses of it anymore